checks.py

```diff
@@ -0,0 +1,414 @@
+        '''
+
+        #: Use existing title unless we have one from metatable
+        if self.new_title:
+            title = self.new_title
+        else:
+            title = self.item.title
+
+        #: Strip off any leading/trailing whitespace
+        orig_tags = [t.strip() for t in self.item.tags]
+
+        #: Add any tags in the metadata to list of tags to evaluate
+        if self.arcpy_metadata and self.arcpy_metadata.tags:
+            orig_tags.extend([t.strip() for t in self.arcpy_metadata.tags.split(', ')])
+
+        #: Evaluate existing tags
+        for orig_tag in orig_tags:
+
+            #: Check if the tag is in the title (checking orig_tag instead
+            #: of cleaned_tag to avoid weird false positives in multi-word
+            #: tags catching the middle of a title- ie, 'Cycle Net' would
+            #: match the title 'Bicycle Network'. Probably not super
+            #: common, but oh well.)
+            #: These combine several boolean checks into a single variable
+            #: to be checked later.
+
+            #: single-word tag in title
+            single_word_tag_in_title = False
+            if orig_tag in title.split():
+                single_word_tag_in_title = True
+            #: multi-word tag in title
+            multi_word_tag_in_title = False
+            if ' ' in orig_tag and orig_tag in title:
+                multi_word_tag_in_title = True
+
+            #: operate on lower case to fix any weird mis-cased tags
+            lowercase_tag = orig_tag.lower()
+
+            #: Run checks on existing tags. A check that modifies the tag should
+            #: append it to new_tags. A check that removes unwanted tags
+            #: should just pass. If a tag passes all the checks, it gets
+            #: properly cased and added to new_tags (the else clause).
+
+            #: Fix/keep 'Utah' if it's not in the title
+            if lowercase_tag == 'utah':
+                #: Have to nest this to avoid 'utah' hitting else and being added
+                if 'Utah' not in title.split():
+                    self.new_tags.append('Utah')
+            #: Don't add to new_tags if it should be deleted
+            elif lowercase_tag in tags_to_delete:
+                pass
+            #: Don't add if it's in the title
+            elif single_word_tag_in_title or multi_word_tag_in_title:
+                pass
+            #: Otherwise, add the tag (properly-cased)
+            else:
+                cased_tag = tag_case(orig_tag, uppercased_tags,  articles)
+                if cased_tag not in self.new_tags:
+                    self.new_tags.append(cased_tag)
+        
+        #: Check the category tag
+        if self.static_shelved == 'shelved':
+            group_tag = 'Shelved'
+        elif self.new_group:  #: If it exists, extract the group
+            group_tag = self.new_group.split('Utah SGID ')[-1]
+        else:  #: If it doesn't exist, set to None
+            group_tag = None
+
```

Comment:
a nice technique to write less lines of code and remove unnecessary branching is to set a default value and then modify that based on the cases. eg:, setting `group_tag = None` initially will remove an else branch in this case. 

If there only 2 cases then you only need one if and no else. simpler code! 

This makes code easier to read and comprehend because there are less branches.

Reply:
I see what you mean about fewer lines of code and branches, and I've changed it for most of the check methods. 

My initial thought was that pure if/elif/else-ing was more readable because every case is contained in the structure, like a switch statement. Using a default is more compact, but might move the default away from the if checks in the code. Thoughts?

Reply:
Typically the default assignment is at the top of the first if branch so it hasn't created mental issues for me.